app/src/main/java/org/mozilla/rocket/download/data/DownloadInfoManager.kt

```diff
@@ -90,8 +90,19 @@ class DownloadInfoManager {
         )
     }
 
-    fun updateByRowId(downloadInfo: DownloadInfo, listener: AsyncUpdateListener?) {
-        mQueryHandler.startUpdate(TOKEN, listener, DownloadContract.Download.CONTENT_URI, getContentValuesFromDownloadInfo(downloadInfo), DownloadContract.Download._ID + " = ?", arrayOf(downloadInfo.rowId.toString()))
+    suspend fun updateByRowId(downloadInfo: DownloadInfo) = suspendCoroutine<Int> { continuation ->
```

Comment:
should we do the query in IO thread?

Reply:
SQL query already runs in the background thread by AsyncQueryHandler.